test/lightning_web/live/work_order_live_test.exs

```diff
@@ -900,6 +900,199 @@ defmodule LightningWeb.RunWorkOrderTest do
     end
   end
 
+  describe "Events" do
+    test "WorkOrders.Events.AttemptCreated", %{
+      conn: conn,
+      project: project
+    } do
+      workflow = insert(:workflow, project: project)
+      trigger = insert(:trigger, type: :webhook, workflow: workflow)
+      job = insert(:job, workflow: workflow)
+
+      dataclip = insert(:dataclip)
+
+      work_order =
+        insert(:workorder,
+          workflow: workflow,
+          trigger: trigger,
+          dataclip: dataclip,
+          attempts: [
+            build(:attempt,
+              dataclip: dataclip,
+              starting_trigger: trigger,
+              runs: [
+                build(:run,
+                  job: job,
+                  exit_reason: "success",
+                  started_at: build(:timestamp),
+                  finished_at: build(:timestamp)
+                )
+              ]
+            )
+          ]
+        )
+
+      {:ok, view, _html} =
+        live(conn, Routes.project_run_index_path(conn, :index, project.id))
+
+      attempt =
+        insert(:attempt,
+          work_order: work_order,
+          dataclip: dataclip,
+          starting_job: job
+        )
+
+      view |> element("#toggle_details_for_#{work_order.id}") |> render_click()
+
+      refute has_element?(view, "#attempt_#{attempt.id}")
+
+      Lightning.WorkOrders.Events.attempt_created(project.id, attempt)
+
+      # Force Re-render to ensure the event is included
+      render(view)
```

Comment:
:+1: yepper, the message is consumed in the LiveView process immediately but the view isn't rendered until you ask it to... under the hood things like `render_click|submit` call render as well.